spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/SkipPatternProviderConfigTest.java

```diff
@@ -97,190 +116,163 @@ public void should_return_empty_when_no_endpoints() {
-				.skipPatternForActuatorEndpointsSamePort(properties,
-						webEndpointProperties, endpointsSupplier)
-				.skipPattern();
-
-		then(pattern).isNotEmpty();
-		then(pattern.get().pattern()).isEqualTo("foo/(info|info/.*|health|health/.*)");
+		contextRunner.withConfiguration(UserConfigurations.of(EndpointConfig.class))
+				.withPropertyValues("management.endpoint.health.enabled=true",
+						"management.endpoint.info.enabled=true",
+						"management.endpoints.web.exposure.include=info,health",
+						"management.endpoints.web.base-path=/",
+						"server.servlet.context-path=foo")
+				.run(context -> {
+					SkipPatternProvider skipPatternProvider = context
+							.getBean(SkipPatternProvider.class);
+					Pattern pattern = skipPatternProvider.skipPattern();
+					then(pattern.pattern())
+							.contains("foo/(health|health/.*|info|info/.*)");
+				});
 	}
 
 	@Test
-	public void should_return_endpoints_with_actuator_context_path_set_to_root() {
-		WebEndpointProperties webEndpointProperties = new WebEndpointProperties();
-		webEndpointProperties.setBasePath("/");
-		ServerProperties properties = new ServerProperties();
-		properties.getServlet().setContextPath("foo");
-
-		EndpointsSupplier<ExposableWebEndpoint> endpointsSupplier = () -> {
-			ExposableWebEndpoint infoEndpoint = createEndpoint("info");
-			ExposableWebEndpoint healthEndpoint = createEndpoint("health");
-
-			return Arrays.asList(infoEndpoint, healthEndpoint);
-		};
-
-		Optional<Pattern> patternDifferentPort = new TraceWebAutoConfiguration.ActuatorSkipPatternProviderConfig()
-				.skipPatternForActuatorEndpointsDifferentPort(properties,
-						webEndpointProperties, endpointsSupplier)
-				.skipPattern();
-
-		then(patternDifferentPort).isNotEmpty();
-		then(patternDifferentPort.get().pattern())
-				.isEqualTo("/(info|info/.*|health|health/.*)");
-
-		Optional<Pattern> patternSamePort = new TraceWebAutoConfiguration.ActuatorSkipPatternProviderConfig()
-				.skipPatternForActuatorEndpointsSamePort(properties,
-						webEndpointProperties, endpointsSupplier)
-				.skipPattern();
-
-		then(patternSamePort).isNotEmpty();
-		then(patternSamePort.get().pattern())
-				.isEqualTo("foo/(info|info/.*|health|health/.*)");
+	public void should_return_endpoints_with_context_path_and_base_path_set_to_root_different_port() {
+		contextRunner.withConfiguration(UserConfigurations.of(EndpointConfig.class))
+				.withPropertyValues("management.endpoint.health.enabled=true",
+						"management.endpoint.info.enabled=true",
+						"management.endpoints.web.exposure.include=info,health",
+						"management.endpoints.web.base-path=/",
+						"management.servet.port=0", "server.servlet.context-path=foo")
+				.run(context -> {
+					SkipPatternProvider skipPatternProvider = context
+							.getBean(SkipPatternProvider.class);
+					Pattern pattern = skipPatternProvider.skipPattern();
+					then(pattern.pattern()).contains("/(health|health/.*|info|info/.*)");
```

Comment:
We have to be careful with using `contains`. It is safest to use `isEqualTo` and just make a constant in this class for the default pattern to concatenate with the interesting portion. I bring this up here because there is a typo in the properties ("servet" → "server") that changes the resulting pattern (/ vs foo/) but the assert as is doesn't catch it.

Reply:
Yeah, this is a tricky part since now we test all patterns together, let me try to find the better way (if any) to ensure we test the right thing.

Reply:
So I did this @shakuzen, I added the assertion on the presence of the exact pattern in the list of `SingleSkipPattern` beans (which `SkipPatternProvider` is built of), it caught the issue with misspelled management port immediately (should have done it from the beginning). Thanks!